website_sale_custom_filter/controllers/main.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2022 Camptocamp
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo.http import request
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class WebsiteSale(WebsiteSale):
+
+    # get min & max from url params for range widgets only
+    def get_min_max_custom_filters(self):
+        req = dict(request.httprequest.args)
+        min_max_filter_vals = {}
+        if req.get("min_cust_filter", False):
+            for vals in req["min_cust_filter"].split(","):
+                f_id, f_min_val = vals.split("_")
+                f_min_val = int(f_min_val)
+                if not min_max_filter_vals.get(f_id, False):
+                    min_max_filter_vals[f_id] = {"min": f_min_val}
+                else:
+                    min_max_filter_vals[f_id]["min"] = f_min_val
+        if req.get("max_cust_filter", False):
+            for vals in req["max_cust_filter"].split(","):
+                f_id, f_max_val = vals.split("_")
+                f_max_val = int(f_max_val)
+                if not min_max_filter_vals.get(f_id, False):
+                    min_max_filter_vals[f_id] = {"max": f_max_val}
+                else:
+                    min_max_filter_vals[f_id]["max"] = f_max_val
+        return min_max_filter_vals
+
+    def get_custom_checkbox_filter_values(self):
+        req = dict(request.httprequest.args)
+        value_filter_data = {}
+        if req.get("cust_filter"):
+            filter_data = req["cust_filter"].split("&")
+            for data in filter_data:
+                f_id, val = data.split("-")
+                if f_id in value_filter_data.keys():
+                    value_filter_data[f_id].append(int(val))
+                else:
+                    value_filter_data[f_id] = [int(val)]
+        return value_filter_data
+
+    def _prepare_filter_actual_website_values(self, min_max_values):
+        values = {}
+        if min_max_values.get("min"):
+            values["min_value"] = min_max_values["min"]
+        if min_max_values.get("max"):
+            values["max_value"] = min_max_values["max"]
+        return values
+
+    def get_custom_range_filter_values(self):
+        range_filter_data = {}
+        cust_filter_min_max_vals = self.get_min_max_custom_filters()
+        curr_website = request.env["website"].get_current_website()
+        numerical_filters_available_on_website = request.env[
+            "website.sale.custom.filter"
+        ].search(
+            [("website_ids", "in", curr_website.id), ("filter_type", "=", "numerical")]
+        )
+
+        for num_filter in numerical_filters_available_on_website:
+            curr_filter_id = str(num_filter.id)
```

Comment:
do you really need this string conversion?

Reply:
we get filter id as part of  string from request so either i cast here as string or after retrieving data from request to int - changed to 2nd option.